### ISSUE TITLE
Why do we need a separate client per job type?

### ISSUE THREAD

**Issue by tobalsgithub** [Automattic/kue#729]
Forgive the possible naivety of this question. We need the ability to have many job types. For each of our customers, we'll need a minimum of 20-30 different job types, and obviously we're hoping to have many customers. We need to be able to individually pause and unpause each of these job types, so we create 1 queue when we bootup the application, and then call `process` for each job type we have.

Recently, we've started hitting ENFILE errors due (we think) to the number of connections that are being created to redis.

I've read #366 and the other issues listed in #535. And I see the code here https://github.com/Automattic/kue/blob/master/lib/queue/worker.js#L267 where we're creating a separate connection per job type.

We're wondering, is there a reason we need a separate client per job type? Is that a redis dictated requirement, like something to do with `blpop`? Is it a stability thing?

Basically, creating a new connection per job type is not going to work for us, so if there's a way to avoid it, we'd be happy to help with a PR. If not, we'll have to look for another solution.

Thanks in advance.

**Comment by behrad** [comment]
This is actualy a redis limitation and that is while the BLPOP command is waiting, the connection is not usable for other commands. So until we are using blocking pop workers we can't share a redis connection!

**Comment by tobalsgithub** [comment]
Makes sense. Thanks for the quick response.

Is using blocking pop workers on the roadmap for kue? Maybe for the 1.0 release?

**Comment by behrad** [comment]
My bad writing @tobalsgithub pordon for the misconception I created, normal job pop operation is mostly implemented via BLPOP or LPOP which Kue has also implemented currently, and since we can't share BLPOP connection.

**Comment by tobalsgithub** [comment]
Ah, ok, so this is just the way it's always going to work?

**Comment by behrad** [comment]
haven't come to a better solution to support distributed work polling for now...
One enhancement however could be a single BLPOP connection to listen on a single redis LIST for all jobs, pop them and dispatch them to its type worker locally.
